contracts/protocol/libraries/logic/LiquidationLogic.sol

```diff
@@ -329,6 +228,203 @@ library LiquidationLogic {
+    DataTypes.ExecuteLiquidationCallParams memory params,
+    LiquidationCallLocalVars memory vars
+  ) internal {
+    if (vars.userVariableDebt >= vars.actualDebtToLiquidate) {
+      vars.debtReserveCache.nextScaledVariableDebt = IVariableDebtToken(
+        vars.debtReserveCache.variableDebtTokenAddress
+      ).burn(
+          params.user,
+          vars.actualDebtToLiquidate,
+          vars.debtReserveCache.nextVariableBorrowIndex
+        );
+    } else {
+      // If the user doesn't have variable debt, no need to try to burn variable debt tokens
+      if (vars.userVariableDebt != 0) {
+        vars.debtReserveCache.nextScaledVariableDebt = IVariableDebtToken(
+          vars.debtReserveCache.variableDebtTokenAddress
+        ).burn(params.user, vars.userVariableDebt, vars.debtReserveCache.nextVariableBorrowIndex);
+      }
+      (
+        vars.debtReserveCache.nextTotalStableDebt,
+        vars.debtReserveCache.nextAvgStableBorrowRate
+      ) = IStableDebtToken(vars.debtReserveCache.stableDebtTokenAddress).burn(
+        params.user,
+        vars.actualDebtToLiquidate - vars.userVariableDebt
+      );
+    }
+  }
+
+  /**
+   * @notice Calculates the total debt of the user and the actual amount to liquidate depending on the health factor
+   * and corresponding close factor.
+   * @dev If the Health Factor is below CLOSE_FACTOR_HF_THRESHOLD, the close factor is increased to MAX_LIQUIDATION_CLOSE_FACTOR
+   * @param debtReserveCache The reserve cache data object of the debt reserve
+   * @param params The additional parameters needed to execute the liquidation function
+   * @param healthFactor the health factor of the position
+   */
+  function _calculateDebt(
+    DataTypes.ReserveCache memory debtReserveCache,
+    DataTypes.ExecuteLiquidationCallParams memory params,
+    uint256 healthFactor
+  )
+    internal
+    view
+    returns (
+      uint256,
+      uint256,
+      uint256
+    )
+  {
+    (uint256 userStableDebt, uint256 userVariableDebt) = Helpers.getUserCurrentDebt(
+      params.user,
+      debtReserveCache
+    );
+
+    uint256 userTotalDebt = userStableDebt + userVariableDebt;
+
+    uint256 closeFactor = healthFactor > CLOSE_FACTOR_HF_THRESHOLD
+      ? DEFAULT_LIQUIDATION_CLOSE_FACTOR
+      : MAX_LIQUIDATION_CLOSE_FACTOR;
+
+    uint256 maxLiquidatableDebt = userTotalDebt.percentMul(closeFactor);
+
+    uint256 actualDebtToLiquidate = params.debtToCover > maxLiquidatableDebt
+      ? maxLiquidatableDebt
+      : params.debtToCover;
+
+    return (userVariableDebt, userTotalDebt, actualDebtToLiquidate);
+  }
+
+  /**
+   * @notice Returns the configuration data for the debt and the collateral reserves.
+   * @param eModeCategories The configuration of all the efficiency mode categories
+   * @param collateralReserve The data of the collateral reserve
+   * @param params The additional parameters needed to execute the liquidation function
+   */
+  function _getConfigurationData(
```

Comment:
the thing is that the closeFactor is not needed anywhere, the only use is to calculate the `actualDebtToLiquidate`, so i'd rather use it here locally - otherwise i need to keep the corresponding field in `LiquidationCallLocalVars`

Reply:
also if i return `closeFactor` from `_getConfigurationData()` i cannot call it after `validateLiquidationCall` as it's needed to calculate the debt

Reply:
natspec fixed